core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -301,6 +301,7 @@ public void onTopologyUpdate(final CacheTopology cacheTopology, final boolean is
       }
       stateTransferLock.releaseExclusiveTopologyLock();
       stateTransferLock.notifyTopologyInstalled(cacheTopology.getTopologyId());
+      remoteCommandsExecutor.checkForReadyTasks();
```

Comment:
haven't we decided that this is bad (and the line 390 too) and we should move it after the command is processed?